src/js/views/patients/patient/sidebar/sidebar_views.js

```diff
@@ -1,142 +1,141 @@
+import { View, CollectionView } from 'marionette';
 
-import intl, { renderTemplate } from 'js/i18n';
 import ModelRender from 'js/behaviors/model-render';
 
-import PatientSidebarTemplate from './patient-sidebar.hbs';
-import PatientSidebarGroupsTemplate from './patient-sidebar-groups.hbs';
-
 import './patient-sidebar.scss';
 import 'sass/domain/engagement-status.scss';
 
-const i18n = intl.patients.patient.sidebar.sidebarViews;
-
-const patientSet = [
-  {
-    heading: i18n.dobLabel,
+const sidebarWidgets = {
+  dob: {
     template: hbs`{{formatHTMLMessage (intlGet "patients.patient.sidebar.sidebarViews.dob") dob=(formatMoment dob "LONG" inputFormat="YYYY-MM-DD") age=age}}`,
-    data(patient) {
-      const dob = patient.get('birth_date');
+    templateContext() {
+      const dob = this.model.get('birth_date');
 
       const age = moment().diff(moment(dob, 'YYYY-MM-DD'), 'years');
 
       return { dob, age };
     },
   },
-  {
-    heading: i18n.sexLabel,
+  sex: {
     template: hbs`{{formatMessage (intlGet "patients.patient.sidebar.sidebarViews.sex") sex=sex}}`,
   },
-];
+  divider: {
+    template: hbs`<div class="patient-sidebar__divider"></div>`,
+  },
+  groups: {
+    behaviors: [
+      {
+        behaviorClass: ModelRender,
+        changeAttributes: ['_groups'],
+      },
+    ],
+    template: hbs`{{#each groups}}<div>{{ this.name }}</div>{{/each}}`,
+    templateContext() {
+      return {
+        groups: _.map(this.model.getGroups().models, 'attributes'),
+      };
+    },
+  },
+  engagement: View.extend({
+    modelEvents: {
+      'change:engagement': 'render',
+      'status:notAvailable': 'onStatusNotAvailable',
+    },
+    getTemplate() {
+      if (!this.model.get('engagement') && !this.notAvailable) {
+        return hbs`<span class="js-loading">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusPreloadView.loading }}</span>`;
+      }
+
+      return hbs`
+        {{#unless notAvailable}}
+          <span class="engagement-status__icon {{ engagement.status }} u-margin--r-4">{{fas "circle"}}</span>{{formatMessage (intlGet "patients.patient.sidebar.sidebarViews.engagementStatusView.status") status=engagement.status}}
+        {{else}}
+          <span class="patient-sidebar__no-engagement">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusView.notAvailable }}</span>
+        {{/unless}}
+      `;
+    },
+    templateContext() {
+      return {
+        notAvailable: this.notAvailable,
+      };
+    },
+    ui: {
+      loading: '.js-loading',
+    },
+    triggers: {
+      'click': 'click',
+    },
+    onRender() {
+      if (!this.model.get('engagement')) {
+        anime({
+          targets: this.ui.loading[0],
+          opacity: 0.5,
+          loop: true,
+          easing: 'easeInOutSine',
+          duration: 400,
+          direction: 'alternate',
+        });
+      }
+    },
+    onClick() {
+      Radio.request('sidebar', 'start', 'engagement', { patient: this.model });
```

Comment:
should this `if (!engagement || isNotAvailable) return;`?  Seems so?

Reply:
Ooh yes. Thanks for catching that.